data/kernels/basic.cl

```diff
@@ -780,14 +780,37 @@ lens_distort_bilinear (read_only image2d_t in, write_only image2d_t out, const i
   const int piwidth = 2*3*width;
   global float *ppi = pi + mad24(y, piwidth, 2*3*x);
 
+  if(!isfinite(ppi[0]) || !isfinite(ppi[1]))
```

Comment:
If this check is true, rx and ry in lines 790 and 791 get "NaN" or "Inf", line 792 assigns a NAN, and line 818 assigns (float4)0.0f.